modules/account-lib/src/coin/sol/utils.ts

```diff
@@ -1,34 +1,76 @@
+import BigNumber from 'bignumber.js';
 
-export class Utils implements BaseUtils {
-  /** @inheritdoc */
-  isValidAddress(address: string): boolean {
-    throw new NotImplementedError('isValidAddress not implemented');
-  }
+const DECODED_BLOCK_HASH_LENGTH = 32; // https://docs.solana.com/developing/programming-model/transactions#blockhash-format
+const DECODED_ADDRESS_LENGTH = 32; // https://docs.solana.com/developing/programming-model/transactions#account-address-format
+const DECODED_SIGNATURE_LENGTH = 64; // https://docs.solana.com/terminology#signature
+const BASE_58_ENCONDING_REGEX = '[1-9A-HJ-NP-Za-km-z]';
 
-  /** @inheritdoc */
-  isValidBlockId(hash: string): boolean {
-    throw new NotImplementedError('isValidBlockId not implemented');
-  }
+/** @inheritdoc */
+export function isValidAddress(address: string): boolean {
+  return isValidPublicKey(address);
+}
 
-  /** @inheritdoc */
-  isValidPrivateKey(key: string): boolean {
-    throw new NotImplementedError('isValidPrivateKey not implemented');
-  }
+/** @inheritdoc */
+export function isValidBlockId(hash: string): boolean {
+  return (
+    !!hash &&
+    new RegExp(BASE_58_ENCONDING_REGEX + '{32,44}').test(hash) &&
+    bs58.decode(hash).length === DECODED_BLOCK_HASH_LENGTH
+  );
+}
 
-  /** @inheritdoc */
-  isValidPublicKey(key: string): boolean {
-    throw new NotImplementedError('isValidPublicKey not implemented');
+/** @inheritdoc */
+export function isValidPrivateKey(prvKey: string | Uint8Array): boolean {
+  const key: Uint8Array = typeof prvKey === 'string' ? this.base58ToUint8Array(prvKey) : prvKey;
+  try {
+    return !!Keypair.fromSecretKey(key);
+  } catch (e) {
+    return false;
   }
+}
 
-  /** @inheritdoc */
-  isValidSignature(signature: string): boolean {
-    throw new NotImplementedError('isValidSignature not implemented');
+/** @inheritdoc */
+export function isValidPublicKey(pubKey: string): boolean {
+  try {
+    return (
+      !!pubKey &&
+      new RegExp(BASE_58_ENCONDING_REGEX + '{32,44}').test(pubKey) &&
+      bs58.decode(pubKey).length === DECODED_ADDRESS_LENGTH &&
+      PublicKey.isOnCurve(new PublicKey(pubKey).toBuffer())
+    );
+  } catch (e) {
+    return false;
   }
+}
 
-  /** @inheritdoc */
-  isValidTransactionId(txId: string): boolean {
-    throw new NotImplementedError('isValidTransactionId not implemented');
-  }
+/** @inheritdoc */
+export function isValidSignature(signature: string): boolean {
+  return !!signature && bs58.decode(signature).length === DECODED_SIGNATURE_LENGTH;
+}
+
+/** @inheritdoc */
+// TransactionId are the first signature on a Transaction
+export function isValidTransactionId(txId: string): boolean {
+  return this.isValidSignature(txId);
+}
+
+/**
+ * Returns whether or not the string is a valid amount of lamports number
+ *
+ * @param {string} amount - the string to validate
+ * @returns {boolean} - the validation result
+ */
+export function isValidAmount(amount: string): boolean {
```

Comment:
For valid amounts exist an Max amount of lamports to send?

Reply:
I havent found a maximum amount but it supports 0 as amount 
https://explorer.solana.com/tx/wASVETEzTgqjsUC3Vb5xkaTBSVSawUkLAmXVY494xACvFN2rUFQ5e3hr1zWTih5rC6NCXXwdFj4htDxdvhBF3xz?cluster=devnet